components/server/src/authorization/permissions.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { v1 } from "@authzed/authzed-node";
+import { inject, injectable } from "inversify";
+import { SpiceDBAuthorizer } from "./spicedb";
+
+import { OrganizationPermission, objectRef, subject } from "./definitions";
+
+export type CheckResult = {
+    permitted: boolean;
+    err?: Error;
+    response?: v1.CheckPermissionResponse;
+};
+
+export const NotPermitted = { permitted: false };
+
+@injectable()
+export class AuthPermissions {
```

Comment:
wraps the generic SpiceDBAuthorizer with domain-specific functions to check permissions in our application code